keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,931 @@
+    |     sources:                                    |
+    |     - secret:                                   |
+    |       name: bucketcreds2                        |
+    |-------------------------------------------------|
+```
+
+The volume `mountPath` will be the directory where bucket credentials and other related information will be served.
+
+NOTE: the contents of the files served in mountPath will be a COSI generated file containing credentials and other information required for accessing the bucket. This is not intended to specify a mountpoint containing the bucket's data.
+
+###### 2. The secret containing BucketInfo is mounted in the specified directory
+
+The above volume definition will prompt kubernetes to retrieve the secret and place it in the volumeMount path defined above. The contents of the secret will be of the format shown below:
+
+```
+    bucket.json
+    |-------------------------------------------------|
+    | {                                               |
+    |   apiVersion: "v1alpha1",                       |
+    |   "authType": "ServiceAccount",                 |
+    |   "S3": {                                       |
+    |     "bucketName": "br-$uuid",                   |
+    |     "endpoint": "https://s3.amazonaws.com",     |
+    |     "region": "us-west-1"                       |
+    |   }                                             |
+    | }                                               |
+    |-------------------------------------------------|
+
+```
+
+Workloads are expected to read the definition in this file to access a bucket. The API of `bucket.json` follows the same versioning and lifecycle as the rest of the COSI APIs.
+
+# Object Lifecycle
+
+The following resources are created by an admin
+
+- BucketClass
+- BucketAccessClass
+
+The following resources are created by a user. They are created with a reference to their corresponding class objects:
+
+- BucketRequest -> BucketClass
+- BucketAccessRequest -> BucketAccessClass, BucketRequest
+
+Note that a BucketAccessRequest also must point to a BucketRequest.
+
+The COSI controller responds by creating the following objects
+
+- BucketRequest <-> Bucket (1:1)
+- BucketAccessRequest <-> BucketAccess (1:1)
+
+In addition, the BucketAccess will point to the Bucket pointed to by the specified BucketRequest
+
+- BucketAccess -> Bucket
+
+Notes:
+
+ - The resulting BucketAccess will point to the same Bucket pointed to by the Bucket Request.
+ - The BucketClass and BucketAccessClass objects are copied into Bucket and BucketAccess respectively. 
+ - There are **NO** cycles in the relationship graph of the above mentioned API objects.
+ - Mutations are not supported in the API.
+
+When a user deletes the BucketRequest, then depending on the DeletionPolicy, the following happens:
+
+- If deletionPolicy is Delete, then Bucket deletion is also triggered. 
+- If deletionPolicy is Retain, then Bucket is left as is, but the BucketRequest is deleted.
```

Comment:
Just to confirm, similar to `brownfield` case can we create BARs from this bucket?

Reply:
This is a great question for existing deployments. I anticipate Rook will have many users interested in this scenario. To my understanding, in the brownfield case, an administrator would create a `Bucket` resource for an existing object store bucket. Then users can create BARs against that Bucket. It would be good to have my understanding validated or corrected.